drivers/npmx/npmx_shell.c

```diff
@@ -1792,6 +1792,160 @@ static int cmd_adc_meas_take_vbat(const struct shell *shell, size_t argc, char *
+		return 0;
+	}
+
+	npmx_led_t *led_instance = npmx_led_get(npmx_instance, led_idx);
+
+	npmx_led_mode_t mode;
+	npmx_error_t err_code = npmx_led_mode_get(led_instance, &mode);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Value: %d.", mode);
+	} else {
+		shell_error(shell, "Error: unable to read LED %d mode.", led_idx);
+	}
+
+	return 0;
+}
+
+static int cmd_leds_mode_set(const struct shell *shell, size_t argc, char **argv)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing LED instance index and LED mode.");
+		return 0;
+	}
+
+	if (argc < 3) {
+		shell_error(shell, "Error: missing LED mode.");
+		return 0;
+	}
+
+	int err = 0;
+	uint8_t led_idx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+	int led_mode = shell_strtol(argv[2], 0, &err);
+
+	if (err != 0) {
+		shell_error(shell, "Error: instance index and mode have to be integers.");
+		return 0;
+	}
+
+	if (led_idx >= NPMX_PERIPH_LED_COUNT) {
+		shell_error(shell, "Error: LED instance index is too high: no such instance.");
+		return 0;
+	}
+
+	npmx_led_t *led_instance = npmx_led_get(npmx_instance, led_idx);
+	npmx_led_mode_t mode;
+
+	switch (led_mode) {
+	case 0:
+		mode = NPMX_LED_MODE_ERROR;
+		break;
+	case 1:
+		mode = NPMX_LED_MODE_CHARGING;
+		break;
+	case 2:
+		mode = NPMX_LED_MODE_HOST;
+		break;
+	case 3:
+		mode = NPMX_LED_MODE_NOTUSED;
+		break;
+	default:
+		shell_error(
+			shell,
+			"Error: LED mode can be 0-Charger error, 1-Charging, 2-HOST, 3-Not used.");
+		return 0;
+	}
+
+	npmx_error_t err_code = npmx_led_mode_set(led_instance, mode);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Success: %d.", led_mode);
+	} else {
+		shell_error(shell, "Error: unable to set LED %d mode.", led_idx);
+	}
+
+	return 0;
+}
+
+static int cmd_leds_state_set(const struct shell *shell, size_t argc, char **argv)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing LED instance index and LED state.");
+		return 0;
+	}
+
+	if (argc < 3) {
+		shell_error(shell, "Error: missing LED state.");
+		return 0;
+	}
+
+	int err = 0;
+	uint8_t led_idx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+	int led_state = shell_strtol(argv[2], 0, &err);
+
+	if (err != 0) {
+		shell_error(shell, "Error: instance index and state have to be integers.");
+		return 0;
+	}
+
+	if (led_idx >= NPMX_PERIPH_LED_COUNT) {
+		shell_error(shell, "Error: LED instance index is too high: no such instance.");
+		return 0;
+	}
+
+	npmx_led_t *led_instance = npmx_led_get(npmx_instance, led_idx);
+
+	npmx_error_t err_code = npmx_led_state_set(led_instance, (led_state ? true : false));
```

Comment:
`led_state` should probably be also checked if it is equal to zero or one before setting the value.